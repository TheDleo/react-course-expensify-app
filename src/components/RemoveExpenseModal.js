import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = (props) => (
    <Modal 
        isOpen={props.modalIsOpen}
        onRequestClose={props.handleCloseModal}
        contentLabel="Remove expense with id:"
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Remove Expense?</h3>
        {props.selectedExpense && <p className="modal__body">{props.selectedExpense.description}</p>}
        <button className="button" onClick={props.onRemove}>Remove It!</button>
        <button className="button button--modal" onClick={props.handleCloseModal}>No!</button>
    </Modal>
);

export default RemoveExpenseModal;