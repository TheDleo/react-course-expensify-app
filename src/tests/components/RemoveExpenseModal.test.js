import React from 'react';
import { shallow } from "enzyme";
import RemoveExpenseModal from "../../components/RemoveExpenseModal";
import expenses from '../fixtures/expenses';

let modalIsOpen, onRemove, handleCloseModal, selectedExpense, wrapper;

beforeEach(() => {
    modalIsOpen = true;
    onRemove = jest.fn();
    handleCloseModal = jest.fn();
    selectedExpense = expenses[1];
    wrapper = shallow(
    <RemoveExpenseModal 
        modalIsOpen={modalIsOpen}
        onRemove={onRemove}
        handleCloseModal={handleCloseModal}
        selectedExpense={selectedExpense}
    />);
});

test('should render RemoveExpenseModal', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should remove expense when remove button clicked', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(onRemove).toHaveBeenCalled();
});

test('should cancel removal when cancel button clicked', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(handleCloseModal).toHaveBeenCalled();
});

test('should render expense name correctly', () => {
    expect(wrapper.find('p .modal__body').text()).toBe(selectedExpense.description);
});