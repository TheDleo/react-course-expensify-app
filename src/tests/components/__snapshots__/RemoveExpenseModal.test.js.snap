// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveExpenseModal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Remove expense with id:"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3
    className="modal__title"
  >
    Remove Expense?
  </h3>
  <p
    className="modal__body"
  >
    Rent
  </p>
  <button
    className="button"
    onClick={[Function]}
  >
    Remove It!
  </button>
  <button
    className="button button--modal"
    onClick={[Function]}
  >
    No!
  </button>
</Modal>
`;
